packages/pangea-sdk/pangea/services/vault/models/common.py

```diff
@@ -129,59 +153,59 @@ class CommonGenerateRequest(APIRequestModel):
     folder: Optional[str] = None
     metadata: Optional[Metadata] = None
     tags: Optional[Tags] = None
-    auto_rotate: Optional[bool] = None
-    rotation_policy: Optional[str] = None
-    store: Optional[bool] = None
+    rotation_frequency: Optional[str] = None
+    rotation_state: Optional[ItemVersionState] = None
     expiration: Optional[datetime.datetime] = None
-    managed: Optional[bool] = None
 
 
 class CommonGenerateResult(PangeaResponseResult):
     type: str
-    version: Optional[int] = None
-    id: Optional[str] = None
+    version: int
+    id: str
 
 
 class GetRequest(APIRequestModel):
     id: str
-    version: Optional[int] = None
+    version: Optional[Union[str, int]] = None
     verbose: Optional[bool] = None
+    version_state: Optional[ItemVersionState] = None
 
 
-class CommonGetResult(PangeaResponseResult):
-    type: str
-    id: str
+class ItemVersionData(PangeaResponseResult):
     version: int
-    name: Optional[str] = None
-    folder: Optional[str] = None
-    metadata: Optional[Metadata] = None
-    tags: Optional[Tags] = None
-    auto_rotate: Optional[bool] = None
-    rotation_policy: Optional[str] = None
-    last_rotated: Optional[str] = None  # TODO: should be time
-    next_rotation: Optional[str] = None  # TODO: should be time
-    expiration: Optional[str] = None  # TODO: should be time
-    created_at: Optional[str] = None  # TODO: should be time
-    revoked_at: Optional[str] = None  # TODO: should be time
-    retain_previous_version: Optional[bool] = None
+    state: str
+    created_at: str
+    destroy_at: Optional[str] = None
+    public_key: Optional[EncodedPublicKey] = None
+    secret: Optional[str] = None
 
 
-class ListItemData(APIRequestModel):
+class ItemData(PangeaResponseResult):
```

Comment:
There are mixed fields in this object from verbose (Item.Type != "folder") with not verbose (Item.Type == "folder").

Reply:
Fixed